src/frontend/src/lib/components/docs/DocForm.svelte

```diff
@@ -0,0 +1,272 @@
+  const { store: docsStore }: DataContext<Doc> = getContext<DataContext<Doc>>(DATA_CONTEXT_KEY);
+
+
+  let state: DataStoreStateEnum | undefined
+  $: state = $docsStore.state
+  let collection: string | undefined;
+	$: collection = $store.rule?.[0];
+
+	let satelliteId: Principal;
+	$: satelliteId = $store.satelliteId;
+
+  let docId: string = '';
+  let fields: DocField[] = [
+    {
+      name: '',
+      fieldType: DocFieldTypeEnum.boolean,
+      value: false,
+    }
+  ]
+
+  const onAddFieldButtonPressed = () => {
+    fields = [
+      ...fields,
+      {
+        name: '',
+        fieldType: DocFieldTypeEnum.boolean,
+        value: false,
+      }
+    ]
+  }
+
+  const onDeleteFieldPressed = (index: number) => {
+    if (fields.length > 1) {
+      fields = fields.filter((_, i) => i !== index)
+    }
+  }
+
+  let isFormValid = false
+  $: isFormValid = !!docId && !!fields.find((field) => !!field.name && !isNullish(field.value) && field.value !== '')
+
+  const dispatch = createEventDispatcher();
+
+  let onSubmit: () => Promise<void>    
+  $: onSubmit = async () => {
+    if (!isFormValid && !collection) return;
+
+    busy.start();
+
+    try {
+      const doc: Record<string, string | number | boolean> = {}
+      fields.forEach((field) => {
+        doc[field.name] = field.value
+      })
+      
+      const docArray = await toArray(doc)
+
+      await setDoc({
+        satelliteId,
+        collection: collection as string,
+        key: docId,
+        doc: {
+          data: docArray,
+          description: toNullable('test description'),
+          updated_at: toNullable(BigInt(new Date().getUTCMilliseconds())),
+        },
+      })
+
+      toasts.success(
+				$i18n.document.document_submission_success
+			);
+
+			await reload();
+
+			dispatch('junoDocumentSubmissionSuccess');
+    } catch (err: unknown) {
+      toasts.error({
+				text: $i18n.document.document_submission_failed,
+				detail: err
+			});
+    } finally {
+      busy.stop();
+    }
+  }
+
+  let isActive = false
+  $: isActive = state === DataStoreStateEnum.CREATE || state === DataStoreStateEnum.EDIT
+
+</script>
+
+<p class="title doc-form">{isActive ? $i18n.document.title_add_new_document : ""}</p>
+
+{#if isActive}
+<article class="doc-form">
+  <form on:submit|preventDefault={onSubmit}>
+    <div>
+      <div>
+        <label>{$i18n.document.field_doc_id_label}</label>
+        <div class="form-doc-id">
+          <input
+            id="doc-id"
+            type="text"
+            placeholder="Document ID"
+            name="doc_id"
+            bind:value={docId}
+          />
+          <button
+            class="primary"
+            on:click|preventDefault={() => docId = generateUniqueDocID()}
+            >
+              {$i18n.document.field_doc_id_btn_auto_id}
+          </button>
+        </div>
+      </div>
+      {#each fields as field, i (i)}
+        <div class="form-field">
```

Comment:
Should we extract this part in a specific component? We can also do it afterwards